src/content/protocols/euler/euler.md

```diff
@@ -0,0 +1,627 @@
+---
+chain: "Ethereum"
+stage: 0
+reasons: []
+risks: ["L", "H", "H", "H", "H"]
+author: ["CookingCryptos"]
+submission_date: "2025-07-01"
+publish_date: "2025-07-01"
+update_date: "2025-07-01"
+stage_requirements:
+  [
+    [
+      {
+        text: "Assets are not in custody by a centralized entity",
+        status: "fixed",
+      },
+      { text: "All contracts are verified", status: "fixed" },
+      { text: "Source-available codebase", status: "fixed" },
+      { text: "Public documentation exists", status: "fixed" },
+    ],
+    [
+      {
+        text: "Upgrades have no potential of “loss of funds“ ",
+        status: "fixed",
+      },
+      {
+        text: "Dependency with a High centralization score is mitigated",
+        status: "fixed",
+      },
+      {
+        text: "Frontend backups or self-hosting option exists",
+        status: "fixed",
+      },
+    ],
+    [
+      {
+        text: "Upgrades have no potential of “loss of funds or unclaimed yield“",
+        status: "fixed",
+      },
+      { text: "There are no external dependencies", status: "fixed" },
+      { text: "Alternative third-party frontends exist", status: "fixed" },
+    ],
+  ]
+---
+
+---
+
+# Summary
+
+Euler V2 is a permissioned, decentralized lending and borrowing protocol deployed on Ethereum. It allows users to deposit assets as collateral, borrow against them, and earn interest via an on-chain interest rate model. The protocol employs smart contracts such as EVaults, a generic factory, and a router to handle asset flows, and integrates with oracles and governance modules to manage risk and updates.
+
+# Ratings
+
+## Chain
+
+Euler V2 is deployed on various chains. This review is based on the Ethereum mainnet deployment.
+
+> Chain score: Low
+
+## Upgradeability
+
+The Euler V2 protocol can be analyzed in a number of logical modules: _Factory-Level Bytecode Upgrades_, _Per-Vault Parameter Control_, _Oracle System_, _Fee Configuration_, _EulerEarn Vault System_, and _EulerSwap_. These modules contain permissioned functions that could result in the _theft or loss of user funds_, _loss of unclaimed yield_, or otherwise _materially affect the expected performance_ of the protocol. The control vectors are distributed across the [DAO](#security-council) multisig, [FactoryGovernor](https://etherscan.io/address/0x2F13256E04022d6356d8CE8C53C7364e13DC1f3d) contract, [Euler Labs](#security-council), vault-specific governors, and various role-based permissions.
```

Comment:
```suggestion
The Euler V2 protocol can be analyzed in a number of logical modules: _Vault Factory_, _Vault Risk Management_, _Oracle System_, _Fee Configuration_, _EulerEarn Vault System_, and _EulerSwap_. These modules contain permissioned functions that could result in the _theft or loss of user funds_, _loss of unclaimed yield_, or otherwise _materially affect the expected performance_ of the protocol. The control vectors are distributed across the [DAO](#security-council) multisig, [FactoryGovernor](https://etherscan.io/address/0x2F13256E04022d6356d8CE8C53C7364e13DC1f3d) contract, [Euler Labs](#security-council), vault-specific governors, and various role-based permissions.
```